run_uh.py

```diff
@@ -294,53 +295,55 @@ def import_fife(paths):
 		return False
 	return True
 
-def find_fife():
-	# Use the path the user provided.
+def get_fife_paths():
+	"""Returns all possible paths to search for the fife module. New paths to be added as needed."""
+	# Use the path the user provided:
 	from horizons.util.cmdlineoptions import get_option_parser
 	options = get_option_parser().parse_args()[0]
+	# All paths collected in a list:
+	paths = []
+	# Check if path was provided by user:
 	if options.fife_path:
 		fife_path = os.path.abspath(options.fife_path)
+		paths.append(fife_path)
 		# Support giving the path to FIFE_ROOT/engine/python/fife/__init__.pyc etc.
+
 		if os.path.isfile(fife_path):
 			fife_path = os.path.dirname(fife_path)
-		# Support giving the path to FIFE_ROOT/engine/python
-		if import_fife([fife_path]):
-			return True
-		# Support giving the path to FIFE_ROOT/engine
-		if import_fife([os.path.join(fife_path, 'python')]):
-			return True
-		# Support giving the path to FIFE_ROOT
-		if import_fife([os.path.join(fife_path, 'engine', 'python')]):
-			return True
-		# Support giving the path to FIFE_ROOT/engine/python/fife
-		if import_fife([os.path.join(fife_path, '..')]):
-			return True
-
-		# End the search to avoid using the wrong (non-user-specified) FIFE.
-		log().error('Unable to find FIFE in %s', fife_path)
-		exit(1)
-
-	# Try to use the default FIFE (equivalent of just trying to import it).
-	if import_fife(None):
-		return True
-
-	# Look for FIFE in the neighborhood of the game dir.
-	paths = []
+			paths.append(fife_path)
+			# Support giving the path to FIFE_ROOT/engine/python
```

Comment:
This was confusing to me in the original code as newlines were extremely sparse. I myself always try to leave space between even the smallest blocks to make the code readable. I'll try to re-arrange this bit properly, though.